src/loaders/workspace_downloader/workspace_downloader.py

```diff
@@ -219,6 +244,17 @@ def process_input(conf):
         with open(metafile, "w", encoding="utf8") as json_file:
             json.dump(_process_object_info(obj_info), json_file, indent=2)
 
+        # remove pre-exsit soft links and reset
+        csd_upa_dir = os.path.join(conf.csd, upa)
+        if os.path.isdir(csd_upa_dir):
+            shutil.rmtree(csd_upa_dir)
+        os.makedirs(csd_upa_dir)
```

Comment:
I am not sure why we want to completely rebuild the csd_upa_dir if they already exist. Shouldn't we check existing link, if not matching, raise error?

Reply:
It does do that I think: https://github.com/kbase/collections/pull/295/files#diff-1ce3e6c7567701005d836ab7d2569c5938318aa586369a8c3224eb495406546aR378-R381

Although if the meta data file is missing or incorrect it just nukes the dir, which to me seems ok

Reply:
I missed that this was the linked dir, not the source dir. You're right AFAICT, we shouldn't be calling rmdir on the link

Reply:
need to remove old csd_upa_dir to softlink csd_upa_dir and dstd, otherwise get file exist error.

Reply:
logic is updated. see below.